tests/integration/security/testdata/authz/v1beta1-jwt.yaml.tmpl

```diff
@@ -59,6 +59,20 @@ spec:
     when:
     - key: request.auth.claims[permission]
       values: ["write"]
+  - to:
+    - operation:
+        paths: ["/nested-key1"]
+        methods: ["GET"]
+    when:
+    - key: request.auth.claims[nested][key1]
```

Comment:
what happens on `request.auth.claims[nested][key1][0]`?

Reply:
and `request.auth.claims[nested][key1][bar]` and `request.auth.claims[nested][not-key1]` and `request.auth.claims[nested]`

Reply:
we don't support indexing in a list so `request.auth.claims[nested][key1][0]` is essentially the same as `request.auth.claims[nested][not-key1]`, it will just not match, same to `request.auth.claims[nested]`.

Added a new test case to cover this explicitly.

Reply:
Do we support three layers of nested claims like request.auth.claims[nested][key1][key2]?